File: java/SnackbarMessage.java
import android.arch.lifecycle.LifecycleOwner;


public class SnackbarMessage extends SingleLiveEvent<String> {

    public void observe(LifecycleOwner owner, final SnackbarObserver observer) {
        super.observe(owner, s -> {
            if (s == null) {
                return;
            }

            observer.onNewMessage(s);
        });
    }

    public interface SnackbarObserver {

        void onNewMessage(String message);
    }
}

File: kotlin/SnackbarMessage.kt
package dev.dita.maziwapp.ui.base.v2

import android.arch.lifecycle.LifecycleOwner
import android.arch.lifecycle.Observer

class SnackbarMessage : SingleLiveEvent<String>() {

    open fun observe(owner: LifecycleOwner, observer: SnackbarObserver) {
        super.observe(owner, Observer {
            if (it == null) {
                return@Observer
            }

            observer.onNewMessage(it)
        })
    }

    open interface SnackbarObserver {
        fun onNewMessage(message: String?)
    }
}